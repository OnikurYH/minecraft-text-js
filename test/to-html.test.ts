import MinecraftText from '..';

describe('toHTML', () => {
  describe('when call with text with color', () => {
    it('should output expected HTML', () => {
        expect(MinecraftText.toHTML('&a&lHello, &rWo&3&m&nrld!\\n&a<Minecraft>&bText&cJS')).toMatchSnapshot();
    });
  });
  describe('when call with unicode text', () => {
    it('should output expected HTML', () => {
        expect(MinecraftText.toHTML('&a&l你好，&r世&3&m&n界!')).toMatchSnapshot();
    });
  });
});
