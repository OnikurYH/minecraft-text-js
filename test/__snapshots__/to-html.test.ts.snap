// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toHTML when call with text with color should output expected HTML 1`] = `"<span style=\\"color: #55FF55;\\"><span style=\\"font-weight: bold;\\">Hello,&nbsp;</span></span>Wo<span style=\\"color: #00AAAA;\\"><span style=\\"text-decoration: line-through;\\"><span style=\\"text-decoration: underline;\\">rld!</span></span></span><br><span style=\\"color: #55FF55;\\">&#60;Minecraft&#62;</span><span style=\\"color: #55FFFF;\\">Text</span><span style=\\"color: #FF5555;\\">JS</span>"`;

exports[`toHTML when call with unicode text should output expected HTML 1`] = `"<span style=\\"color: #55FF55;\\"><span style=\\"font-weight: bold;\\">&#20320;&#22909;ï¼Œ</span></span>&#19990;<span style=\\"color: #00AAAA;\\"><span style=\\"text-decoration: line-through;\\"><span style=\\"text-decoration: underline;\\">&#30028;!</span></span></span>"`;
